src/vs/platform/theme/browser/iconsStyleSheet.ts

```diff
@@ -31,34 +31,41 @@ export function getIconsStyleSheet(themeService: IThemeService | undefined): IIc
 		getCSS() {
 			const productIconTheme = themeService ? themeService.getProductIconTheme() : new UnthemedProductIconTheme();
 			const usedFontIds: { [id: string]: IconFontDefinition } = {};
-			const formatIconRule = (contribution: IconContribution): string | undefined => {
+
+			const rules: string[] = [];
+			const rootAttribs: string[] = [];
+			for (const contribution of iconRegistry.getIcons()) {
 				const definition = productIconTheme.getIcon(contribution);
 				if (!definition) {
-					return undefined;
+					continue;
 				}
+
 				const fontContribution = definition.font;
+				const fontFamilyVar = `--vscode-icon-${contribution.id}-font-family`;
+				const contentVar = `--vscode-icon-${contribution.id}-content`;
 				if (fontContribution) {
 					usedFontIds[fontContribution.id] = fontContribution.definition;
-					return `.codicon-${contribution.id}:before { content: '${definition.fontCharacter}'; font-family: ${asCSSPropertyValue(fontContribution.id)}; }`;
-				}
-				// default font (codicon)
-				return `.codicon-${contribution.id}:before { content: '${definition.fontCharacter}'; }`;
-			};
-
-			const rules = [];
-			for (const contribution of iconRegistry.getIcons()) {
-				const rule = formatIconRule(contribution);
-				if (rule) {
-					rules.push(rule);
+					rootAttribs.push(
+						`${fontFamilyVar}: ${asCSSPropertyValue(fontContribution.id)};`,
+						`${contentVar}: '${definition.fontCharacter}';`,
+					);
+					rules.push(`.codicon-${contribution.id}:before { content: '${definition.fontCharacter}'; font-family: ${asCSSPropertyValue(fontContribution.id)}; }`);
```

Comment:
The font-family property can be overridden by the more specific rule `.monaco-workbench .predefined-file-icon[class*=codicon-]:before`.
I found two icons that have their font-family property overridden by this rule , those being the `search` product icon (rule `.codicon-search-editor-label-icon:before`) when used as tab icon for a search editor tab, and the `settings` product icon (rule `.codicon-settings-editor-label-icon:before`) in a settings editor tab.

Reply:
This PR does not try to change the priority of the existing CSS rules